core/eth/state.go

```diff
@@ -59,7 +85,131 @@ func (cs *ChainState) GetOperatorSocket(ctx context.Context, blockNumber uint, o
 	return socket, nil
```

Comment:
This method, `GetOperatorSocket`, doesn't take `blockNumber` into account.

I know this wasn't changed in this PR, but I'm wondering if this is a latent bug

Reply:
Seems like we are just logging `blockNumber` when there's an error, and the interface for ChainState is used as part of `IndexedChainState` which uses the `blockNumber` in its version of `GetOperatorSocket`. I think we should keep the `blockNumber` for `IndexedChainState`, but could assume the latest state for reading from `ChainState`, or even create an variant of it